apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -50,16 +57,38 @@ func (host *BareMetalHost) validateChanges(old *BareMetalHost) []error {
 	return errs
 }
 
-func validateRAID(r *RAIDConfig) error {
-	if r == nil {
-		return nil
+func validateBMCAccess(s BareMetalHostSpec, bmcAccess bmc.AccessDetails) []error {
+	var errs []error
+
+	if bmcAccess == nil {
+		return errs
 	}
 
-	if len(r.HardwareRAIDVolumes) > 0 && len(r.SoftwareRAIDVolumes) > 0 {
-		return fmt.Errorf("hardwareRAIDVolumes and softwareRAIDVolumes can not be set at the same time")
+	if s.RAID != nil {
+		if bmcAccess.RAIDInterface() == "no-raid" {
+			errs = append(errs, fmt.Errorf("the 'raid' field for %s is not supported", bmcAccess.Type()))
```

Comment:
For consistency with other error messages: "BMC driver %s does not support configuring RAID".

By the way, all this logic is completely broken for software RAID :-/ I guess we need to solve it eventually.

Reply:
Actually, no, only this validation is broken. You need to check for hardware RAID configuration, for software RAID we will set RAIDInterface to "agent" when handling it.

Reply:
Thanks, updated. PTAL.